drivers/retained_mem/Kconfig.nrf

```diff
@@ -13,6 +13,5 @@ config RETAINED_MEM_NRF_RAM_CTRL
 	bool "nRF RAM retention driver"
 	default y
 	depends on DT_HAS_ZEPHYR_RETAINED_RAM_ENABLED && RETAINED_MEM_ZEPHYR_RAM && POWEROFF
-	depends on SOC_SERIES_NRF52X || SOC_SERIES_NRF53X || SOC_SERIES_NRF54LX || SOC_SERIES_NRF91X
```

Comment:
this needs a depends on for nrf socs

Reply:
added dependency on `SOC_FAMILY_NORDIC_NRF`